internal/helm/helm.go

```diff
@@ -122,47 +145,53 @@ func makeStack(appname string, targetDir string, data []byte, stackVersion strin
 	if err := os.MkdirAll(filepath.Join(targetDir, "templates"), 0755); err != nil {
 		return err
 	}
-	var stack interface{}
+	var stackData []byte
 	switch stackVersion {
 	case V1Beta2:
 		stackSpec := templateconversion.FromComposeConfig(rendered)
-		stack = templatev1beta2.Stack{
-			TypeMeta:   typeMeta(stackVersion),
+		stack := templatev1beta2.Stack{
+			TypeMeta: templatev1beta2.TypeMeta{
+				Kind:       "Stack",
+				APIVersion: "compose.docker.com/" + stackVersion,
+			},
 			ObjectMeta: objectMeta(appname),
 			Spec:       stackSpec,
 		}
+		stackData, err = convertTemplatesAny(stack)
+		if err != nil {
+			return err
+		}
 	case V1Beta1:
-		composeFile, err := yaml.Marshal(rendered)
+		composeFile, err := convertTemplatesAny(rendered)
 		if err != nil {
 			return err
 		}
-		stack = v1beta1.Stack{
+		stack := v1beta1.Stack{
 			TypeMeta:   typeMeta(stackVersion),
 			ObjectMeta: objectMeta(appname),
 			Spec: v1beta1.StackSpec{
 				ComposeFile: string(composeFile),
 			},
 		}
+		// Kube doesn't have the right annotations for YAML, so serialize into json
```

Comment:
@mnottale any more precision on this ? `cli/compose` types are serializable in YAM… `v1beta1.Stack` (or `v1beta2:Stack` should be too, no ? :thinking:

Reply:
everything is good except for the TypeMeta which comes from kube.